bundles/org.openhab.core.addon/src/main/java/org/openhab/core/addon/AddonMatchProperty.java

```diff
@@ -0,0 +1,41 @@
+/**
+ * Copyright (c) 2010-2023 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.core.addon;
+
+import java.util.regex.Pattern;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+
+/**
+ * DTO for serialization of a property match regular expression.
+ *
+ * @author Andrew Fiddian-Green - Initial contribution
+ */
+@NonNullByDefault
+public class AddonMatchProperty {
+    private @NonNullByDefault({}) String name;
+    private @NonNullByDefault({}) Pattern pattern;
```

Comment:
Make this field transient, or you end up with a GSON exception as pattern can't be serialized.
I would keep the String regex field as well, so the REST endpoint does get the regex.

Reply:
Good point.